src/odemis/driver/pigcs.py

```diff
@@ -638,18 +641,18 @@ def GetAvailableParameters(self):
 
     def GetParameter(self, axis, param):
         """
-        axis (bytes): axis number
+        axis (str): axis number
         param (0<int): parameter id (cf p.35)
         returns (str): the string representing this parameter
         """
         # SPA? (Get Volatile Memory Parameters)
-        assert(isinstance(axis, bytes) and 1 <= len(axis) <= 8)
```

Comment:
Please don't drop this check (and all the other ones). Just update it bytes -> str.

Reply:
len(axis) is always 1, so it doesn't really make sense to test for that

Reply:
also we should use basestring, if we decode senquerycommand all the time, then some commands can be unicode

Reply:
What makes you think that `axis` always of length 1? I'm pretty sure that the user could pass any string as axis. Also, I think that the hardware supports axes names of multiple characters. See `GetAxes()`.